src/axom/quest/examples/quest_step_file.cpp

```diff
@@ -0,0 +1,2223 @@
+      for(int i = 0; i < vals.size(); ++i)
+      {
+        knots.insert(knots.end(), mults[i], vals[i]);
+      }
+      SLIC_ASSERT(knots.size() == total_knots);
+
+      return knots;
+    }
+
+    /// converts knot values to axom Array w/o accounting for multiplicity
+    /// /sa extractKnotMultiplicities
+    axom::Array<double> extractKnotValues() const
+    {
+      const int num_knots = m_curve->NbKnots();
+      axom::Array<double> knots(0, num_knots);
+
+      TColStd_Array1OfReal occ_knots(1, num_knots);
+      m_curve->Knots(occ_knots);
+      for(int i = 1; i <= occ_knots.Length(); ++i)
+      {
+        knots.push_back(occ_knots(i));
+      }
+
+      return knots;
+    }
+
+    /// converts knot multiplicities to axom Array
+    /// /sa extractKnotValues
+    axom::Array<int> extractKnotMultiplicities() const
+    {
+      const int num_knots = m_curve->NbKnots();
+      axom::Array<int> mults(0, num_knots);
+
+      TColStd_Array1OfInteger occ_mults(1, num_knots);
+      m_curve->Multiplicities(occ_mults);
+      for(int i = 1; i <= occ_mults.Length(); ++i)
+      {
+        mults.push_back(occ_mults(i));
+      }
+
+      return mults;
+    }
+
+  private:
+    Handle(Geom2d_BSplineCurve) m_curve;
+    bool m_verbose {false};
+    bool m_inputSurfaceWasPeriodic {false};
+  };
+
+public:
+  StepFileProcessor() = delete;
+
+  StepFileProcessor(const std::string& filename, bool verbose = false)
+    : m_verbose(verbose)
+  {
+    m_shape = loadStepFile(filename);
+  }
+
+  void setVerbosity(bool verbose) { m_verbose = verbose; }
+
+  bool isLoaded() const { return m_loadStatus == LoadStatus::SUCEESS; }
+
+  const TopoDS_Shape& getShape() const { return m_shape; }
+
+  int getNumberOfPatches() const { return m_patchData.size(); }
+
+  /// Logs some information about the loaded mesh
+  void printMeshInfo() const
+  {
+    // Helper struct for simple stats over a collection of integers
+    struct AccumStatistics
+    {
+      int min;
+      int max;
+      double mean;
+      double stddev;
+    };
+
+    // Lambda to generate AccumStatistics for a list of integers
+    auto computeStatistics = [](const std::vector<int>& data) -> AccumStatistics {
+      AccumStatistics stats;
+      stats.min = *std::min_element(data.begin(), data.end());
+      stats.max = *std::max_element(data.begin(), data.end());
+
+      const double sum = std::accumulate(data.begin(), data.end(), 0.0);
+      const double sumSquared =
+        std::accumulate(data.begin(), data.end(), 0.0, [](double a, double b) {
+          return a + b * b;
+        });
+      stats.mean = sum / data.size();
+      stats.stddev =
+        std::sqrt(sumSquared / data.size() - stats.mean * stats.mean);
+      return stats;
+    };
+
+    axom::fmt::memory_buffer out;
+
+    axom::fmt::format_to(std::back_inserter(out),
+                         "Details about loaded mesh:\n");
+
+    // summarize the number of patches and trimming curves
+    {
```

Comment:
There are several internal scopes here that I am not sure why. Where they intentional?

Reply:
It was intentional -- I wanted to separate the different steps and scope their internal variables